webapp/channels/src/components/sidebar/channel_navigator/channel_navigator.tsx

```diff
@@ -37,14 +37,18 @@ export default class ChannelNavigator extends React.PureComponent<Props> {
         document.removeEventListener('keydown', this.handleQuickSwitchKeyPress);
     }
 
+    private buttonRef = React.createRef<HTMLButtonElement>();
+
     openQuickSwitcher = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
         e.preventDefault();
+        const focusOriginElement = this.buttonRef;
 
         trackEvent('ui', 'ui_sidebar_open_channel_switcher_v2');
 
         this.props.actions.openModal({
             modalId: ModalIdentifiers.QUICK_SWITCH,
             dialogType: QuickSwitchModal,
+            dialogProps: {focusOriginElement},
```

Comment:
from here, the originElement is referenced and pass as a dialogProp to the modal.